### PR TITLE
fix: standalone ajv schema

### PR DESCRIPTION
**Pull request by climba03003**
This PR do not need to merge immediately.
We need to test each edge case before merging this PR.

1. It remove `fast-json-stringify` in `standalone` code.
2. It address the issue of `fast-json-stringify` generate custom json schema.
3. It update the types.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by climba03003** [comment]
Given that people complain https://github.com/fastify/fastify/issues/3998 it about the `fast-json-stringify/serialize` apporach.
I see the point of @ivan-tymoshenko is absolutely right.

This patch will remove the dependent of `fast-json-stringify` internal files.

File: README.md [hunk 1]
```diff
@@ -679,8 +679,8 @@ console.log(stringify({ firstName: 'Foo', surname: 'bar' })) // '{"firstName":"F
 ### Standalone Mode
 
 The standalone mode is used to compile the code that can be directly run by `node`
-itself. You need to install `fast-json-stringify`, `ajv`, `fast-uri` and `ajv-formats`
-in order to let the standalone code works.
+itself. You need to install `ajv`, `fast-uri` and `ajv-formats` in order to let 
+the standalone code works.
```
Comment (Fdawgs) [inline README.md:683]:
```suggestion
itself. You need to install `ajv`, `fast-uri` and `ajv-formats` for
the standalone code to work.
```

**Review by Eomm — APPROVED** [review]

File: index.d.ts [hunk 2]
```diff
@@ -157,14 +157,38 @@ declare namespace build {
      * Optionally configure how the integer will be rounded
      */
     rounding?: 'ceil' | 'floor' | 'round'
+    /**
+     * @depercated
```
Comment (mcollina) [inline index.d.ts:161]:
```suggestion
     * @deprecated 
```

**Review by mcollina — COMMENTED** [review]
Good work, one nit

File: standalone.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  const serializerCode = fs.readFileSync(path.join(__dirname, 'serializer.js')).toString()
+  const buildAjvCode = fs.readFileSync(path.join(__dirname, 'ajv.js')).toString()
+  let defaultAjvSchema = ''
+  const defaultMeta = ajvInstance.defaultMeta()
+  if (typeof defaultMeta === 'string') {
+    defaultAjvSchema = defaultMeta
+  } else {
+    defaultAjvSchema = defaultMeta.$id || defaultMeta.id
+  }
+  console.log(defaultAjvSchema)
```
Comment (climba03003) [inline standalone.js:14]:
```suggestion

```

**Comment by mcollina** [comment]
lmk when it's ok to release

**Comment by ivan-tymoshenko** [comment]
@climba03003 Do we need to generate a standalone serializer only here?
https://github.com/fastify/fastify/blob/main/build/build-error-serializer.js

**Comment by mcollina** [comment]
I would not exclude we could need to do it elsewhere too.

**Review by climba03003 — COMMENTED** [review]
@mcollina 
I think it is ok to land now.
One final tweak is that it should not write `ajv` to standalone script when it is not needed.

File: index.d.ts [hunk 4]
```diff
@@ -157,14 +157,38 @@ declare namespace build {
...
+     */
+    mode?: 'debug' | 'standalone'
   }
 }
 
+interface DebugOption extends build.Options {
+  mode: 'debug'
+}
+
+interface DepercateDebugOption extends build.Options {
```
Comment (zedeus) [inline index.d.ts:176]:
```suggestion
interface DeprecateDebugOption extends build.Options {
```

File: index.d.ts [hunk 5]
```diff
@@ -157,14 +157,38 @@ declare namespace build {
...
+  mode: 'standalone'
+}
+
 /**
  * Build a stringify function using a schema of the documents that should be stringified
  * @param schema The schema used to stringify values
  * @param options The options to use (optional)
  */
+declare function build(schema: build.AnySchema, options: DebugOption): { code: string, ajv: Ajv };
+declare function build(schema: build.AnySchema, options: DepercateDebugOption): { code: string, ajv: Ajv };
```
Comment (zedeus) [inline index.d.ts:190]: Here as well
```suggestion
declare function build(schema: build.AnySchema, options: DeprecateDebugOption): { code: string, ajv: Ajv };
```

**Comment by mcollina** [comment]
@climba03003 could you send the PRs needed to fastify and https://github.com/fastify/fast-json-stringify-compiler? Please update the versions everywhere.
